ui/src/main/java/edu/wpi/grip/ui/pipeline/input/InputSocketController.java

```diff
@@ -59,6 +62,11 @@ protected void initialize() {
     this.type.setText(this.socket.getSocketHint().getTypeLabel());
     this.handle = socketHandleViewFactory.create(this.socket);
     root.add(this.handle, 0, 0);
+    handle.connectedProperty().addListener((observable, oldValue, newValue) -> {
+      if (getContent() != null) {
+        getContent().setDisable(newValue);
```

Comment:
It's currently implemented in each class (that's a lt of repetition). This just puts it in the common superclass to a) use a listener instead of a binding and b) let me programmatically enable and disable every socket